operator/v1/types_network.go

```diff
@@ -119,6 +120,25 @@ type NetworkSpec struct {
 	// migration procedure allows to change the network type and the MTU.
 	// +optional
 	Migration *NetworkMigration `json:"migration,omitempty"`
+
+	// AdvancedRouting enables support for additional routing capabilities, like
+	// BGP. When set, it triggers the rollout of the necessary components,
+	// including but not limited to FFR-k8s, enabling the use of these routing
+	// capabilities for the machine network. Upstream operators, like MetalLB
+	// operator, requiring these capabilities may rely or automatically set this
+	// flag. Network plugins may either not leverage any advanced routing
+	// capability related to this setting, may honor this configuration without
+	// any aditional specific configuration or, like ovn-kubernetes, require
+	// additional enablement through their own specific configuration. Refer to
+	// their respective documentation and configuration options. Allowed values
+	// are "Enabled", "Disabled" and ommited. When omitted, this means the user
+	// has no opinion and the platform is left to choose reasonable defaults.
+	// These defaults are subject to change over time. The current default is
+	// "Disabled".
+	// +openshift:enable:FeatureGate=BGP
+	// +kubebuilder:validation:Enum="";Enabled;Disabled
+	// +optional
+	AdvancedRouting string `json:"advancedRouting,omitempty"`
```

Comment:
This is a configuration API so we wouldn't want `omitempty` here, it will enable discovery of the fields when users are configuring the API